test/hotspot/jtreg/compiler/loopopts/superword/TestMemorySegment.java

```diff
@@ -0,0 +1,860 @@
+ */
+
+/*
+ * @test id=long-array
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment LongArray
+ */
+
+/*
+ * @test id=float-array
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment FloatArray
+ */
+
+/*
+ * @test id=double-array
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment DoubleArray
+ */
+
+/*
+ * @test id=byte-buffer
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment ByteBuffer
+ */
+
+/*
+ * @test id=byte-buffer-direct
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment ByteBufferDirect
+ */
+
+/*
+ * @test id=native
+ * @bug 8310190
+ * @summary Test vectorization of loops over MemorySegment
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.TestMemorySegment Native
+ */
+
+// FAILS: mixed providers currently do not vectorize. Maybe there is some inlining issue.
+// /*
+//  * @test id=mixed-array
+//  * @bug 8310190
+//  * @summary Test vectorization of loops over MemorySegment
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment MixedArray
+//  */
+//
+// /*
+//  * @test id=MixedBuffer
+//  * @bug 8310190
+//  * @summary Test vectorization of loops over MemorySegment
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment MixedBuffer
+//  */
+//
+// /*
+//  * @test id=mixed
+//  * @bug 8310190
+//  * @summary Test vectorization of loops over MemorySegment
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment Mixed
+//  */
+
+public class TestMemorySegment {
+    public static void main(String[] args) {
+        TestFramework framework = new TestFramework(TestMemorySegmentImpl.class);
+        framework.addFlags("-DmemorySegmentProviderNameForTestVM=" + args[0]);
+        framework.setDefaultWarmup(100);
+        framework.start();
+    }
+}
+
+class TestMemorySegmentImpl {
+    static final int BACKING_SIZE = 1024 * 8;
+    static final Random RANDOM = Utils.getRandomInstance();
+
+
+    interface TestFunction {
+        Object[] run();
+    }
+
+    interface MemorySegmentProvider {
+        MemorySegment newMemorySegment();
+    }
+
+    static MemorySegmentProvider provider;
+
+    static {
+        String providerName = System.getProperty("memorySegmentProviderNameForTestVM");
+        provider = switch (providerName) {
+            case "ByteArray"        -> ( () -> { return newMemorySegmentOfByteArray(); } );
```

Comment:
You can directly use an expression lambda without return: 
```
case "ByteArray"        -> (() -> newMemorySegmentOfByteArray());
```
But I think you can go even further and directly use a method reference:
```suggestion
            case "ByteArray"        -> (TestMemorySegmentImpl::newMemorySegmentOfByteArray);
```
Same for others.